ocaml/rrdd/rrdd_server.ml

```diff
@@ -628,12 +628,18 @@ module Plugin = struct
 			verify_checksum_freshness ~uid ~checksum;
 			parse_payload ~json:payload
 		with e ->
-			error "Failed to process plugin: %s (%s)" uid (Printexc.to_string e);
-			log_backtrace ();
+			let log e =
+				error "Failed to process plugin: %s (%s)" uid (Printexc.to_string e);
+				log_backtrace ()
+			in
 			match e with
-			| Invalid_header_string | Invalid_length | Invalid_checksum
-			| No_update as e -> raise e
-			| _ -> raise Read_error
+			| No_update -> raise No_update
```

Comment:
All exceptions get handled [here](https://github.com/xenserver/xen-api/blob/creedence-bugfix/ocaml/rrdd/rrdd_server.ml#L681-684) - if read_file raises anything then the mainloop will catch it silently and move on to the next plugin.